src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -492,6 +492,26 @@ parseNumberBinder = NumberBinder <$> (sign <*> number)
          <|> (symbol' "+" >> return id)
          <|> return id
 
+parseTypedBinder :: TokenParser Binder
+parseTypedBinder = do
+  b <- typeableBinder
+  indented *> doubleColon
+  t <- parseType
+  return $ TypedBinder t b
+    where typeableBinder = P.choice (map P.try
```

Comment:
Here is one place, where types with a kind annotation get parsed:

https://github.com/purescript/purescript/blob/master/src/Language/PureScript/Parser/Types.hs#L103

I was wrong, the function you want is probably `buildPostfixParser`.